### ISSUE TITLE
Community management concerns

### ISSUE THREAD

**Issue by atiertant** [balderdashy/waterline#1208]
We doesn't like the way the community is being run.  Sailsbot is an annoying symptom of a worse underlying problem that bugs and features are being ignored or unacknowledged for extended periods of time.
As collaborators, community members, and business users of Waterline, we need to know that bugs and features will be treated appropriately.  Maintaining a fork to integrate our features and fixes is not preferable, but is becoming increasingly necessary.

**Comment by devinivy** [comment]
I do generally agree that progress has been slow, and that sailsbot's activity is indicative of a greater problem.  Having just written the rest of this github comment, I apologize for its length.  But there's much to say and clarify!  For those not familiar with me, I'm just an interested collaborator on the waterline project.

From @clarkorz's famously impressive and old denormalization PRs (e.g. https://github.com/balderdashy/sails-mongo/pull/149) to @atiertant's more recent deep-populate PR (https://github.com/balderdashy/waterline/pull/1052), not to mention his roadmap PR (https://github.com/balderdashy/waterline/pull/1196), there's a lot sitting around (also, @rwoverdijk's applied dirty-checking https://github.com/balderdashy/waterline/pull/1048 and some others)!  And those are just a few features! Waterline spent a fair amount of time playing catchup (big kudos to the core team and especially @dmarcelino) to get tests, issues, and roadmap planning in order, but after that there has been a struggle to move forward.  There are countless bugs, features, and performance concerns that feel important but forgotten, some being reopened twice after auto-closes by sailsbot.

Sailsbot is an oversimplified solution to a difficult problem– waterline comprises a large ecosystem made possible by busy people who are part of what can only be assumed to be a successful OSS-contributing company based around node (Balderdashy).  Over the past few months, this has meant that many waterline modules (including waterline itself) have not received the upkeep that they live up to.  As mentioned before, pending bugs and features abound.  I see why sailsbot is a desirable solution to this.  But it's _really_ not the right solution.  Sailsbot has caused frustration within the community.  This includes core members, collaborators, major contributors, and casual waterline users– essentially, the full gamut.  I'm sure I don't need to cite explicit examples here to make that point; I feel certain that active community members already know this.  I've reopened countless confirmed bugs and also features that had been spec'd then placed into milestones.  The policy propagated by sailsbot that feature requests should be made as PRs to roadmap.md does not work well if existing feature requests are not migrated to roadmap PRs and such PRs that are made (e.g. #1196) are not acknowledged.

I don't think the issue here is one of lack of effort (on the part of core members) or ungratefulness (on the part of the larger community).  I enjoy being part of his community because the technology is exciting and fun to work with, and the Balderdash crew is top-notch.  @tjwebb, @particlebanana, @mikermcneil, and several other core contributors have written and maintained a metric crapton of code.  It is highly appreciated– this is free, (I'll be redundant) open-source software, after all, and ultimately (despite pending bugs and features) it works pretty well.  This is complex stuff!  Nobody is forcing us to use it!   Turns out it's fun to use!

The main reason this issue exists is because the way the community is being managed does not work well for waterline's users.  I doubt anyone would prefer a fork.  That would suck.  I doubt it will be necessary.  But if the project will otherwise stand still, then even if just for business reasons, I can see why maintaining a fork might be seen as worthwhile or the only safe long-term solution for certain users– I understand why @atiertant mentions that option.  I understand why there's a lot of frustration around long-pending issues.

Waterline is an ambitious project that requires many hands on deck.  I think if we organize existing collaborators and distribute responsibility in a thoughtful way, we'll find that this community/maintenance problem is solvable.  I'm very interested to hear from the core team on this, and I look forward to generating a solution.  I do think that this is a case where assigning roles and adding some light formality to community management might be in order.  I'm a huge fan of the project, Balder, and the community members, and I'll continue to be around to help.

**Comment by leobudima** [comment]
@atiertant, thank you for bringing up this topic, and @devinivy, thanks for an extensive and thoughtful comment. Although I have participated in the community on only a few occasions, I too believe, based on what I've seen, that the problem is not in lack of people willing to help - maybe just to get organized a bit better and spread the load. 

That said,  we have a Sails / Waterline app in fairly successful start of production, and I'd be interested and willing to help and invest time in getting some of the issues handled and PRs integrated. My personal view is that people would, at this moment and given the rather nice working feature set of Sails / Waterline, get more benefit from ironing out bugs and integrating PRs, than from new features.

So if we get to discuss this further and start (re)organizing around collaboration, I'm happy to say you can count me in.

**Comment by RWOverdijk** [comment]
I'm with @leobudima on this one; but that's not a secret. You know where to find me?

**Comment by BLamy** [comment]
In the past I have always used express for small projects that didn't need a database. As soon as I needed a database I'd generate a sails app. Without waterline I'll probably just stick with express from here on out.

**Comment by sailsbot** [comment]
Thanks for posting, @atiertant.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by RWOverdijk** [comment]
The fact that sailsbot closes this is hilariously ironic.

**Comment by mikermcneil** [comment]
@RWOverdijk then again, it was an open issue that was no longer under discussion.

**Comment by randallmeeker** [comment]
@mikermcneil The fact that you chose to defend @sailsbot instead of addressing this issue raised @atiertant  in the comment speaks to what @RWOverdijk was addressing.

**Comment by tjwebb** [comment]
> was no longer under discussion.

According to who? A robot? Your personal opinion? If you ask the actual participants in the discussion -- @atiertant, @devinivy, @RWOverdijk, @BLamy, @leobudima -- I suspect you'll get a different response.

Whatever, it's now closed, and per usual, @mikermcneil's only response is to defend his closing of it, rather than to respond substantively to the actual, real issue raised here. The Sails King has declared your **"Community management concerns"** to be not worth discussing. Move along, nothing to see here.

**Comment by RWOverdijk** [comment]
@mikermcneil Right... You're on fire. :)

**Comment by atiertant** [comment]
this is true that it was no longer in discussion because discussion never started...
for this reasons i started a fork of waterline called [offshore](https://github.com/Atlantis-Software/offshore)

**Comment by RWOverdijk** [comment]
@atiertant Love the enthusiasm, but you should first fork a working version (including the links and docs etc) before announcing it. Now people will bounce off.

**Comment by atiertant** [comment]
@RWOverdijk npm install offshore
this is working, need some docs that's true...
wouldn't like to announce it now but i think it's now the in discussion (ironicly after sailsbot close the discussion)

**Comment by leobudima** [comment]
Well, based on the current state of sails and waterline and from recent discussions, I can't see any other reasonable path than to move on to something else...I mean, come on - on sails-mongo, there are PRs (that are still passing CI) dating from 2014. Is it that hard to accept contributor's help if you can't merge a few PRs in a year? A simple issue of auto-reconnecting to the database - which is critical for serious production, and not a FR - has not been solved in 20 MONTHS. And why would anyone bother to submit a PR, when they can only hope for it to be accepted by 2017?

I think I'll try to find time and rewrite our server without Sails...Continuing to develop further with a dead framework can't lead anywhere nice. Very much looking forward to Trails and Offshore, I'll contact you guys to offer help as soon as I clear the current mess on our side.

**Comment by RWOverdijk** [comment]
@leobudima I just want to point out, before this escalates any further, that things are moving again. Sails is not dead, and neither is waterline.

**Comment by particlebanana** [comment]
@leobudima we have been looking for a maintainer for sails-mongo for years. If you are using it and are interested let me know.

**Comment by Nishchit14** [comment]
@leobudima 

I am using sails-mongo since last 2 year. And i am interested in further development in it.
I have faced many problems regarding grouping &  other advance aggregation query , so it would fine to take action in this era.

**Comment by mikermcneil** [comment]
@Nishchit14 would be glad for the help :+1:

**Comment by mikermcneil** [comment]
I hate to bring this up again, since I imagine everyone is pretty done hearing about this by now, but I'm still getting some questions from folks who haven't seen the other places this is posted.  In particular, this issue has been linked to a number of times alongside some of the more misleading claims that have been made about this project (e.g. from `sails-permissions`, a community plugin).  For the benefit of folks arriving here from those links, or via search engines, I want to be sure I've tied up all the loose ends. 

I've posted more about this in the [Status of Sails](https://github.com/balderdashy/sails/issues/3429#issuecomment-168801414) issue, but heres a quick recap:

+ Trails is not affiliated with Sails or our team.
+ Only one person left the Sails.js project to create Trails: Travis Webb; someone who joined the Sails project fairly recently as a [contributor](https://github.com/balderdashy/sails/graphs/contributors) in 2015.
+ The [core team](https://github.com/balderdashy/sails#team) and I are 100% committed to the continued development and support of Sails and Waterline.  We currently use the latest stable release of both libraries in production, as do many other companies and developers all over the world.

If you are using `sails-permissions` or another plugin in your project: and you have immediate strategic or ethical concerns that you don't feel comfortable posting publicly;  please [contact me directly](mailto:mission@treeline.io).  Otherwise if your question is technical in nature, and you feel it would help others, please open an issue in [Sails' GitHub repository](http://github.com/balderdashy/sails).  Normally, the issue queue in the Sails repo is not the place for questions about 3rd party plugins, but given the circumstances I'm willing to make an exception.

On a related note, one fuzzy bit of silver lining to come out of all these events was an official [Code of Conduct](https://github.com/balderdashy/sails/blob/master/CODE-OF-CONDUCT.md) for the Sails framework and related projects.  Please give that a quick read if you haven't already.  It clarifies the bare minimum behavior expected from participants in this community, as well as some baseline recommendations for effective communication with faceless strangers on the internet :)

Finally, as I've stated elsewhere: If you have questions or concerns about Sails, Waterline, [our team](https://github.com/balderdashy/sails#team), what we're working on, or where we're headed, please don't hesitate to [reach out to me directly](mailto:mission@treeline.io).

-"The Sails King"
